### ISSUE TITLE
Setting name of TTree that is different from cbmsim

### ISSUE THREAD

**Issue by ThomasRuf** [FairRootGroup/FairRoot#1111]
Our experiment SND@LHC is starting recording real data. I would like to give the name of the TTree a name different from cbmsim, more related to real data. The only way to change the default name at the moment is to create a directory config, adding a file with the content, "treename=name_you_chose". BUT this is not a good  solution, since I want to be able to read real and MC data in parallel. Don't want to have different projects. Ideally, there should be a set method of the FairRootManager to set the default name. My current solution, create this file on the fly before calling FairFileSource and delete it after. This is not a perfect solution, since it will fail if two jobs, one for MC and one for data are starting at the same time. 
Any suggestions are welcome. Thomas

**Comment by dennisklein** [comment]
Thank you very much for the suggestion. We are looking into it.

**Comment by karabowi** [comment]
Dear Thomas,

we have discussed the issue on our weekly meeting
and we propose you to create your own SndFileSource and SndRootFileSink.
There, the created/read tree is set its name.

In the corresponding FairFileSource and FairRootFileSink, the names
are taken from FairRootManager via the GetTreeName() function.
In your implementations, you will be able to set the tree name 
dynamically to whatever you like.

Should you have any problem, contact us!

Best regards,
Radek

**Comment by hnatics** [comment]
Is it possible to redesign current implementation at some point? 

if I must create a folder /config in main with a one-liner file in it, this is quite a bad design...

**Comment by ChristianTackeGSI** [comment]
Can we close this, now that #1140 has landed?

@ThomasRuf can you take a look, whether #1140 looks useful to you?

**Comment by ThomasRuf** [comment]
What I coincidence! I just started to modify FairRootManager to have this functionality. I first thought also about a Set method, but was then thinking about gROOT->GetListOfGlobals()->FindObject("TreeName") instead of reading from a file.  But if a Set method is now available, will immediately use it. Thank you very much.

**Comment by ThomasRuf** [comment]
Was a bit too optimistic with testing. Just changing to the FairRoot dev version, and building in our environment with alibuild immediately fails, see below. We are currently using  version v18.4.2. What would be the fastest / efficient way to implement #1140 ? 

CMake Error at CMakeLists.txt:95 (find_package):
  By not providing "FindFairCMakeModules.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "FairCMakeModules", but CMake did not find one.

**Comment by ChristianTackeGSI** [comment]
I have opened PRs to backport this to 18.6 (#1151) and 18.4 (#1152). Maybe that helps?

@karabowi What do you think?

**Comment by ThomasRuf** [comment]
I tried to build with just the changes to FairRootManager. It works, except for, auto f = fMan->GetOutFile(); However, the related piece of code is not needed for snd@lhc, so, I disabled it. What would I have to do to get the modification from  
source: https://github.com/FairRootGroup/FairRoot ? Currently, I am using tag: "v18.4.2"

**Comment by ChristianTackeGSI** [comment]
I don't understand, how this is related to `auto f = fMan->GetOutFile();`? Where is this code?

I don't know, how to integrate this in your build system, to be honest. The PRs are getting merged into the `v18.6_patches` and `v18.4_patches` branches and sometime later there might be releases. At least, that's my understanding of the release procedure.

**Comment by ThomasRuf** [comment]
OK, just let me know when I can pickup the patched v18.4 version.

Regarding the issue I mentioned above:
FairRootManager of v18.4 had a method: TFile* GetOutFile(); This method disappeared in the dev version. One older piece of code in our software was using this method. 
Maybe only merging the changes regarding SetTreeName would eventually not give a problem.

**Comment by ChristianTackeGSI** [comment]
The `GetOutFile` method has been deprecated since 18.0.

Commit deprecating it: 1da8249a8fbf07380bb7b2f5017c809a61cf2fe9 and d7fc6ab2817ac96b06fe712f56921d4db9282ed5

It has been removed in the dev branch, which is going to become 19.0 at some point.
PR for the removal: #1127

You should look into rewriting that part of your code.

**Comment by ChristianTackeGSI** [comment]
JFI: The patch is now in the v18.4_patches branch as well.
